dart/lib/src/http_client/tracing_client.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:http/http.dart';
+import '../hub.dart';
+import '../hub_adapter.dart';
+import '../protocol.dart';
+
+/// A [http](https://pub.dev/packages/http)-package compatible HTTP client
+/// which adds support to Sentry Performance feature.
+/// https://develop.sentry.dev/sdk/performance
+class TracingClient extends BaseClient {
+  TracingClient({Client? client, Hub? hub})
+      : _hub = hub ?? HubAdapter(),
+        _client = client ?? Client();
+
+  final Client _client;
+  final Hub _hub;
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    // see https://develop.sentry.dev/sdk/performance/#header-sentry-trace
+    final span = _hub.getSpan()?.startChild(
+          'http.client',
+          description: '${request.method} ${request.url}',
+        );
+
+    StreamedResponse? response;
+    try {
+      if (span != null) {
+        final traceHeader = span.toSentryTrace();
+        request.headers[traceHeader.name] = traceHeader.value;
+      }
+
+      // TODO: tracingOrigins support
+
+      response = await _client.send(request);
+      span?.status = SpanStatus.fromHttpStatusCode(response.statusCode);
+    } catch (exception) {
+      span?.throwable = exception;
+      span?.status = SpanStatus.internalError();
+
+      rethrow;
+    } finally {
+      await span?.finish();
```

Comment:
I wonder if this should be awaited. 

It would make the users HTTP call potentially take longer just because we're potentially sending a trace to Sentry. 

Also if the BreadcrumClient is around this it would skew the recorded time of it.

What do you think?